### ISSUE TITLE
Inbound ICD calls show as Type Outbound

### ISSUE THREAD

**Issue by l4pul4pu** [bdm1981/finesseGadgets#34]
Great gadget mate...

Couple things that I've noticed, and not sure if I've done something wrong, or have something misconfigured somewhere.
 - Inbound ICD calls show as type "Outbound" and the number is the not the original caller number but rather the UCM Trigger number.
 - Inbound calls that ring out and forward to Voicemail show up with 2 "Inbound" Call entries

**Comment by bdm1981** [comment]
Hi @l4pul4pu 

Are you running this in CCE or UCCX and what version?

Do you have voicemail enabled on your agents lines? Voicemail on the agent extension isn't really a supported configuration.

**Comment by l4pul4pu** [comment]
Apologies, more details would have been handy.

UCM 10.5.2
UCCX 10.6
Firefox 58.0.1
as 3rdpartygadget

Our agents aren't just dedicated call center staff, so they also receive direct calls to their extensions, which is the only time calls can get to their voicemail.  We use Exchange UM for voicemail

ICD calls never reach their voicemail.

The "Inbound" calls showing twice (when they reach voicemail) are NON-ICD calls direct to the extension.

Is it possible to differentiate ICD vs NON-ICD inbound calls, and have both as separate entries on the Call Type Table?

I still don't understand how inbound ICD calls are showing up as Outbound calls though.

**Comment by bdm1981** [comment]
Thanks for the additional detail. Can you send me the Firefox developer console logs from both of these call scenarios? I will spin up my lab and see if I can reproduce them.

**Comment by l4pul4pu** [comment]
Well..  I'm now at a loss as to all my concerns raised here..

Started to gather the console logs, and strangely enough, everything is working as intended.

I did not get the double entry for direct calls to the extension (even when forward to voicemail).

I got the correct inbound call showing now for ICD calls when it was showing as outbound previously.

I think I've either lost my mind or loosing it slowly.

Thanks for your help though.  I think we're all good, must've been noob error on my part.

Works awesome, great work mate.  Lets close off this issue.
